app/src/main/java/cafetcraftteam/rpgsoftware/CharacterWarhammer.java

```diff
@@ -37,18 +41,214 @@
             /*
     Equipment
      */
-    private Map mWeapons;
-    private Map mArmor;
-    private Map mDefensePoints;
-    private Map mEquipment;
+    private HashMap<Integer,String> mWeapons;
+    private HashMap<Integer,String> mArmor;
+    private HashMap<Bodypart,Integer> mDefensePoints;
+    private HashMap<Integer,String> mEquipment;
 
             /*
     Abilities and Talents
      */
 
-    private Map mBasicAbilities;
-    private Map mAdvanAbilities;
+    private HashMap<String,String> mBasicSkills;
+    private HashMap<String,String> mAdvanSkills;
+
+    CharacterWarhammer(String Name, String RealName, String Campaign, String Year, String Race, int Age, String Sex, String Eyecolor, int Height, String Haircolor, int Weight, String SA, int NbBrSis, String BirthPlace, String DistinctifSign){
+        super(Name, RealName, Campaign, Year, Race);
+        mAge = Age;
+        mSex = Sex;
+        mEyecolor = Eyecolor;
+        mHeight = Height;
+        mHaircolor = Haircolor;
+        mWeight = Weight;
+        mSA = SA;
+        mNbBrSis = NbBrSis;
+        mBirthPlace = BirthPlace;
+        mDistinctifSign = DistinctifSign;
+        mDefensePoints.put(Bodypart.HEAD, 0);
+        mDefensePoints.put(Bodypart.TORSO, 0);
+        mDefensePoints.put(Bodypart.LEFTARM, 0);
+        mDefensePoints.put(Bodypart.RIGHTARM, 0);
+        mDefensePoints.put(Bodypart.LEFTLEG, 0);
+        mDefensePoints.put(Bodypart.RIGHTLEG, 0);
+        mBasicSkills.put("Animal Care", "Skill");
+        mBasicSkills.put("Charm", "Skill");
+        mBasicSkills.put("Command", "Skill");
+        mBasicSkills.put("Concealment", "Skill");
+        mBasicSkills.put("Consume Alcohol", "Skill");
+        mBasicSkills.put("Disguise", "Skill");
+        mBasicSkills.put("Drive", "Skill");
+        mBasicSkills.put("Evaluate", "Skill");
+        mBasicSkills.put("Gamble", "Skill");
+        mBasicSkills.put("Gossip", "Skill");
+        mBasicSkills.put("Haggle", "Skill");
+        mBasicSkills.put("Intimidate", "Skill");
+        mBasicSkills.put("Outdoor Survival", "Skill");
+        mBasicSkills.put("Perception", "Skill");
+        mBasicSkills.put("Ride", "Skill");
+        mBasicSkills.put("Row", "Skill");
+        mBasicSkills.put("Scale Sheer Surface", "Skill");
+        mBasicSkills.put("Search", "Skill");
+        mBasicSkills.put("Silent Move", "Skill");
+        mBasicSkills.put("Swim", "Skill");
+    }
+
+    public String getmProfession() {
+        return mProfession;
+    }
+
+    public void setmProfession(String mProfession) {
+        this.mProfession = mProfession;
+    }
+
+    public Integer getPrimaryProfile(Char C) {
+        return mPrimaryProfile.get(C);
+    }
+
+    public void setPrimaryProfile(Char C, int Value) {
+        this.mPrimaryProfile.put(C, Value);
```

Comment:
`this` est inutile. Et au passage changer tes noms de variables local dans les autres fonctions permettrait de le supprimer aussi.